writer_downgrade_test.go

```diff
@@ -235,15 +250,63 @@ func (s *WriterDowngradeSuite) validateRead(b *bytes.Buffer) {
 		s.Assert().Nil(err)
 		s.Assert().Equal(32.99, math.Round(price*100)/100)
 
-		// Advance to the second array element
-		err = r.AdvanceToNextElement(buf)
-		s.Assert().Nil(err)
+		err = r.AdvanceTo(buf, "products", "variations")
+		if err != ErrNoSuchField {
+			s.Assert().Nil(err)
+
+			// Read the first array element's "variations" array index
+			//
+			// full array size
+			_, err = r.ReadSizeField(buf)
+			s.Assert().Nil(err)
+			//
+			// Array length should be two
+			arrayLen, err = r.ReadSizeField(buf)
+			s.Assert().Nil(err)
+			s.Assert().Equal(2, arrayLen)
+			//
+			// Array index
+			// Entry 1
+			id, err := r.ReadIntField(buf)
+			s.Assert().Nil(err)
+			s.Assert().Equal(int64(9), id)
+			_, err = r.ReadSizeField(buf)
+			s.Assert().Nil(err)
+			//
+			// Entry 2
+			id, err = r.ReadIntField(buf)
+			s.Assert().Nil(err)
+			s.Assert().Equal(int64(11), id)
+			_, err = r.ReadSizeField(buf)
+			s.Assert().Nil(err)
+			//
+			// Array elements
+			err = r.AdvanceTo(buf, "products", "variations", "description")
+			s.Assert().Nil(err)
+			desc, err := r.ReadStringField(buf)
+			s.Assert().Nil(err)
+			s.Assert().Equal("variation one", desc)
+			//
+			// Next element
+			err = r.AdvanceToNextElement(buf)
+			s.Assert().Nil(err)
+			//
+			err = r.AdvanceTo(buf, "products", "variations", "description")
+			s.Assert().Nil(err)
+			desc, err = r.ReadStringField(buf)
+			s.Assert().Nil(err)
+			s.Assert().Equal("variation two", desc)
+
+			// Advance to the array end
+			err = r.AdvanceToNextElement(buf, "products", "")
+			s.Assert().Nil(err)
+		}
 
 		// Get the second array element's "Name" field
 		err = r.AdvanceTo(buf, "products", "name")
 		s.Assert().Nil(err)
 		name, err = r.ReadStringField(buf)
-		s.Assert().Nil(err)
+		s.Require().Nil(err)
```

Comment:
Why just change this to `Require()`?

Reply:
Oh, I changed this while testing to reduce the number of errors I was seeing. I can never decide which to use. On one hand, it's nice to use assertions so you can see all the errors at once. On the other hand, sometimes its easier to fix errors one at a time, in which case requirements are nice. Have you figured out a good pattern?

Reply:
Yeah I struggle with that as well. I usually use `Require()` when the rest of the test is dependent on the results of what comes before it, but I haven't been consistent one way or another. I agree that I usually prefer to fix one issue at a time, it can be overwhelming to see a bunch of errors at once.

Reply:
For now, I just reverted this one.